src/vivarium/framework/population.py

```diff
@@ -178,7 +178,12 @@ def get(self, index: pd.Index, query: str = "") -> pd.DataFrame:
             non_existent_columns = set(columns) - set(pop.columns)
             if non_existent_columns:
                 raise PopulationError(
-                    f"Requested column(s) {non_existent_columns} not in population table."
+                    f"Requested column(s) {non_existent_columns} not in population table. This is"
+                    f" likely due to a failure to require some columns, randomness streams, or "
+                    f" pipelines when registering a simulant initializer, a value producer, or a"
+                    f" value modifier. NOTE: it is possible for this not to fail even if"
+                    f" requirements were not specified and then randomly fail later if the"
+                    f" initialization order changes."
```

Comment:
This still seems a little unclear to me. Maybe: "It is possible for a run to succeed even if resource requirements were not properly specified in the simulant initializers or pipeline creation/modification calls.  This success depends on component initialization order which may change in different run settings."

Reply:
+1 for this wording --- its a bit easier to follow